### DISCUSSION TITLE
DragAndDrop with CustomNode work question

### DISCUSSION THREAD

**Discussion by kimchijoa** [bcakmakoglu/vue-flow#1305] [Q&A]
Hello
I'm using vueflow well.

Questions arise about functionality while in use.

First, I'm thinking of the following functions.
1. Create a CustomNode.
2. Configure CustomNode to be generated as DragAndDrop.
3. If you click on the CustomNode you created, you can change Style.

The implementation of 1 and 2 was completed, but there was a problem in number 3.

CustomNodes using templates had the following issues.
1. To dynamically give className, find the NodeID using the querySelector after DragAndDrop and change the ClassName.
2. When a new CustomNode is added, nodes of the same type as the added CustomNode are initialized as initial information (both className and Label)

CustomNode was used for image insertion and resizing in Node.

In this situation, if you implement CustomNodes through templates and create CustomNodes through DragAndDrop, it leaves you wondering if it is technically natural for the same type of CustomNodes to be initialized.

I'll leave the code I used below. Thank you.

Leave github link because the code is too long.
```
1. main vue - src/views/VueNodeTest.vue
2. CustomNode vue - src/views/CustomNode.vue
3. Sidebar vue(for DragAndDrop) - src/views/SidebarNode.vue
```

**Comment by kimchijoa** [comment]
sry, this is my github link
https://github.com/kimchijoa/VueFlow_Prototype

**Comment by bcakmakoglu** [comment]
Not sure what the issue exactly is - what *isn't* working?

If you need to add a class name to a node you can use `node.class = 'my-class-name'`. This can be done *per node*.

**Comment by wk-iei** [comment]
@kimchijoa hi, I have meet the same problem, have you fixed this ?
